third-party/realdds/include/realdds/dds-stream.h

```diff
@@ -25,12 +36,25 @@ class dds_stream : public dds_stream_base
 protected:
     dds_stream( std::string const & stream_name, std::string const & sensor_name );
 
-    // dds_stream_base
 public:
-    bool is_open() const override;
-    bool is_streaming() const override;
+    bool is_open() const override { return !! _reader; }
+    virtual void open( std::string const & topic_name, std::shared_ptr< dds_subscriber > const & );
+    virtual void close();
+
+    bool is_streaming() const override { return _is_streaming; }
+    typedef std::function< void( const topics::device::image & f) > on_data_available_callback;
+    void start_streaming( on_data_available_callback cb );
+    void stop_streaming();
 
     std::shared_ptr< dds_topic > const & get_topic() const override;
+
+protected:
+    void handle_frames();
+
+    std::shared_ptr< dds_topic_reader > _reader;
+    on_data_available_callback _on_data_available;
+    dispatcher _image_dispatcher;

```

Comment:
Not good. Let's talk about it.

Reply:
My concerns are these:
* Too many dispatchers. FastDDS docs already state that they create a thread "for each reception cannel". Then there's the librealsense dispatchers...
* I'm not sure we should worry about this at the realdds layer. Why not just let the callback decide where to push it to, what to do with it, etc.? I.e., even if we want to do something time-consuming, we know this only at the application-level, and putting it here is assuming a lot...
* The dispatcher is a complicated beast designed for access from multiple threads... a simple thread with a queue might be all that's needed. But let's talk about this once we've resolved the question of where this belongs...

Reply:
Removed dispatcher